source_http.go

```diff
@@ -100,6 +109,10 @@ func newHTTPRequest(s *HTTPImageSource, ireq *http.Request, method string, url *
 		s.setAuthorizationHeader(req, ireq)
 	}
 
+	if len(s.Config.CustomHeaders) != 0 {
+		s.setCustomHeaders(req, ireq)
```

Comment:
This sequence opens up a vulnerability regarding the Authorization Header. As effectively this allows the client to override the Authorization Header.

I'm not sure what's wise here:

- Change the sequence, so that adding Imaginary's own headers is always last
- Change the implementation to include a `if ! ireq.Header.Has(header) {`

I think my preference would be to do both.